### PR TITLE
fix: ChartGrid flicker and zoom reset when new data

### PR DESCRIPTION
**Pull request by trentwatt**
## Description

[WEB-848]

Before:
```
 data is updated
-> chartsProps changes
-> <ChartGrid> rerenders because it takes chartsProps as a prop
-> the renderer for the individual grid items gets redefined, 
   since it is an inline anonymous function within the <ChartGrid> component
-> all the individual grid items get rerendered
-> the remounting of `UPlotChart` wrapper component causes the chart to be destroyed and created
-> flicker
```

After: 
- the renderer is its own component: no remount, no flicker
- fixed the way charts are initialized if there is already synced min/max. this fixes the zoom reset issue even without the other fix, making it more robust.
- got rid of `<AutoSizer>` which is designed to be used with `react-virtualized`, but we are using `react-window` for virtualization. replaced it with the `useResize` hook we use elsewhere. allows us to get rid of another anonymous function without having to add additional component.

## Test Plan

Most of the changes here only pertain to chart grid, need to just check single trial with `f_chart=on` and design kit chart grid. [this](https://github.com/determined-ai/determined/files/10772007/test-metrics-grid.zip) is a good experiment for testing, creates a long running experiment with frequent updates.

## Checklist

- [X] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-848]: https://determinedai.atlassian.net/browse/WEB-848?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ClaireNeveu** [comment]
Since this is a requirement of the component, could you add an example to the UI Kit with updating data?

File: webui/react/src/components/kit/LineChart.tsx [hunk 1]
```diff
@@ -250,11 +250,34 @@ interface GroupProps {
   xAxis: XAxisDomain;
 }
 
+const VirtualChartRenderer: React.FC<
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/LineChart.tsx:253]: Probably worth documenting why this is necessary.

**Comment by trentwatt** [comment]
> Since this is a requirement of the component, could you add an example to the UI Kit with updating data?

will do!

File: webui/react/src/components/kit/LineChart.tsx [hunk 2]
```diff
@@ -1,6 +1,5 @@
-import React, { useMemo, useState } from 'react';
-import AutoSizer from 'react-virtualized-auto-sizer';
```
Comment (mapmeld) [inline webui/react/src/components/kit/LineChart.tsx:2]: Note that dropping this = you can remove `react-virtualized-auto-sizer` and `@types/react-virtualized-auto-sizer` from package.json

**Review by mapmeld — APPROVED** [review]
Tested chart in UI Kit, LGTM ✅

**Review by gt2345 — APPROVED** [review]
LGTM, chart grid in UI kit looks smooth.
Also it would be nice to remove "react-virtualized-auto-sizer" like Nick said to reduce bundle size

**Pull request merged**